pizza_cutter/des_pizza_cutter/_se_image.py

```diff
@@ -0,0 +1,448 @@
+                )
+                image_pos = self._wcs.toImage(world_pos)
+                x.append(image_pos.x - 1)
+                y.append(image_pos.y - 1)
+            x = np.array(x)
+            y = np.array(y)
+        else:
+            raise ValueError('WCS %s not recognized!' % self._wcs)
+
+        if is_scalar:
+            return x[0], y[0]
+        else:
+            return x, y
+
+    def get_wcs_jacobian(self, x, y):
+        """Return the Jacobian of the WCS transformation as a
+        `galsim.JacobianWCS` object.
+
+        Parameters
+        ----------
+        x : scalar or 1d array-like
+            The x/column image location in zero-indexed, pixel centered
+            coordinates.
+        y : scalar or 1d array-like
+            The y/row image location in zero-indexed, pixel centered
+            coordinates.
+
+        Returns
+        -------
+        jac : galsim.JacobianWCS
+            The WCS Jacobian at (x, y).
+        """
+        assert np.ndim(x) == 0 and np.ndim(y) == 0, (
+            "WCS Jacobians are only returned for a single position at a time")
+
+        if isinstance(self._wcs, eu.wcsutil.WCS):
+            tup = self._wcs.get_jacobian(x+1, y+1)
+            dudx = tup[0]
+            dudy = tup[1]
+            dvdx = tup[2]
+            dvdy = tup[3]
+            jac = galsim.JacobianWCS(
+                dudx, dudy, dvdx, dvdy)
+        elif isinstance(self._wcs, galsim.BaseWCS):
+            pos = galsim.PositionD(x=x+1, y=y+1)
+            jac = self._wcs.local(image_pos=pos)
+        else:
+            raise ValueError('WCS %s not recognized!' % self._wcs)
+
+        return jac
+
+    def contains_radec(self, ra, dec):
+        """Use the approximate sky bounds to detect if the given point in
+        ra, dec is anywhere in the total SE image (not just the slice).
+
+        Parameters
+        ----------
+        ra : scalar or 1d array-like
+            The right ascension of the sky position.
+        dec : scalar or 1d array-like
+            The declination of the sky position.
+
+        Returns
+        -------
+        in_sky_bnds : bool or boolean array mask
+            True if the point is in the approximate sky bounds, False
+            otherwise.
+        """
+        if np.ndim(ra) == 0 and np.ndim(dec) == 0:
+            is_scalar = True
+        else:
+            is_scalar = False
+
+        assert np.ndim(ra) <= 1 and np.ndim(dec) <= 1, (
+            "Inputs to sky2mage must be scalars or 1d arrays")
+        assert np.ndim(ra) == np.ndim(dec), (
+            "Inputs to sky2image must be the same shape")
+
+        ra = np.atleast_1d(ra).ravel()
+        dec = np.atleast_1d(dec).ravel()
+
+        u, v = radec_to_uv(ra, dec, self._ra_ccd, self._dec_ccd)
+        in_sky_bnds = self._sky_bnds.contains_points(u, v)
+
+        if is_scalar:
+            return bool(in_sky_bnds[0])
+        else:
+            return in_sky_bnds
+
+    def get_psf_image(self, x, y):
```

Comment:
OK @esheldon. I'd start here looking through the PSF rendering code. I left notes in the comments.